test/sys/test_socket.rs

```diff
@@ -255,3 +255,62 @@ pub fn test_syscontrol() {
     // requires root privileges
     // connect(fd, &sockaddr).expect("connect failed");
 }
+
+// Test creating and using named system control sockets
+#[cfg(all(target_os = "linux", target_arch = "x86_64", not(target_env = "musl")))]
+#[test]
+pub fn test_scm_timestamping() {
+    use nix::sys::uio::IoVec;
+    use nix::unistd::close;
+    use nix::sys::socket::{socket, bind, connect, sendmsg, recvmsg, setsockopt,
+                           AddressFamily, SockType, SockFlag, SockAddr,
+                           CmsgSpace, MsgFlags, ControlMessage};
+    use nix::sys::socket::sockopt::{Timestamping};
+    use nix::sys::time::TimeVal;
+    use std::net;
+
+    let actual: net::SocketAddr = FromStr::from_str("127.0.0.1:3000").unwrap();
```

Comment:
port 3000 is already used by a different test.  Can you choose a different port?